app/src/main/java/com/codesoom/assignment/application/UserService.java

```diff
@@ -0,0 +1,43 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.domain.User;
+import com.codesoom.assignment.dto.UserData;
+import org.springframework.stereotype.Service;
+
+/**
+ * 사용자 생성, 수정, 삭제 기능을 담당하는 클래스
+ */
+@Service
+public interface UserService {
+
+    /**
+     * 사용자를 생성하고, 생성된 사용자 정보를 리턴합니다.
+     * @param source 생성할 사용자 정보
+     * @return 생성된 사용자를 반환한다.
+     */
+    User createUser(UserData source) throws Exception;
+
+    /**
+     * 사용자 정보를 수정하고 수정된 사용자 정보를 리턴합니다.
+     * @param id 사용자 id
+     * @param source 수정할 사용자 정보
+     * @return 수정된 사용자를 반환한다.
+     */
+    User updateUser(Long id, UserData source);
+
+    /**
+     * 사용자를 삭제합니다.
+     * @param id 삭제할 사용자 id
+     */
+    void deleteUser(Long id);
+
+    /**
+     * 이메일 중복확인
+     * @oaram 중복 확인용 메일
+     * @return 중복있음 - true / 중복없음 - false
+     * @throws Exception 중복이 있으면 UserEmailDuplicateException 예외를 던진다.
+     */
+    boolean emailCheck(String mail) throws Exception;
```

Comment:
좀 이상합니다. `@return`을 읽으면 이메일이 중복되었다면 true를 리턴한다고 설명이 있는데요, `@throws`를 읽으면 중복이 있다면 예외를 던진다고 합니다. 둘 중 어느쪽인가요?

```suggestion
    /**
     * 이메일이 중복되었다면 true, 그렇지 않다면 false를 리턴합니다.
     *
     * @param 중복을 확인하려는 이메일 주소
     * @return 중복 이메일이 존재한다면 true, 그렇지 않다면 false
     * @throws UserEmailDuplicateException 이메일이 중복된 경우
     */
    boolean emailCheck(String mail) throws UserEmailDuplicateException;
```

Reply:
헷갈릴 만한 부분인 것 같아요 ! 이러한 부분은 더 상세히 작성하도록 하겠습니다 !